web/src/compose_notifications.ts

```diff
@@ -89,6 +90,10 @@ export function notify_automatic_new_visibility_policy(
 function get_message_header(message: Message): string {
     if (message.type === "stream") {
         const stream_name = stream_data.get_stream_name_from_id(message.stream_id);
+        if (message.topic === "") {
+            const topic_display_name = util.get_final_topic_display_name(message.topic);
+            return `#${stream_name} > <span class="empty-topic-display">${topic_display_name}</span>`;
+        }
```

Comment:
This would introduce an XSS security bug. (Or more technically, changing to inserting this into the DOM via `{{{`, which skips escaping).

But a general code smell that should immediately indicate to you this code can't be correct is that you can never mix user input (stream/topic names) with HTML without escaping and expect that to not introduce an XSS vulnerability. 

I'm not sure what the best option is; one good one is probably to make a simple Handlebars template for displaying a stream/topic pair correctly into HTML, passing the appropriate variables.  

But one thing to be careful about is that it is a bit suspicious to have a function where some code paths return a plain-text string, and others return HTML.